Add support for CMS AOD

sudo-panda opened the pull request:
- Update code to support changes to func_adl_xAOD
- Add conf file for CMS AOD
- Update Dockerfile to add support for build-args
- Update ci.yaml to publish CMS AOD Dockerfile

Things to check before merging:

* [x] the xAOD code-gen works fine
* [ ] the CMS AOD code-gen works fine

sudo-panda committed:
Add support for CMS AOD

- Update code to support changes to func_adl_xAOD
- Add conf file for CMS AOD
- Update Dockerfile to add support for build-args
- Update ci.yaml to publish CMS AOD Dockerfile

gordonwatts commented on requirements.txt:
```diff
@@ -3,4 +3,4 @@ Flask==1.1.2
 Flask-WTF==0.14.3
 
-func-adl-xAOD.backend==1.0.0b5
+git+git://github.com/sudo-panda/func_adl_xAOD.git@setup-py-fix#egg=func_adl_xAOD
```
We don't want to check this into the repo...

gordonwatts replied:
Ok - as draft - but we need to make sure we do not actually let this into the final repo

gordonwatts replied:
Updated to be the beta release

gordonwatts commented on servicex/code_generator_service/ast_translator.py:
```diff
@@ -48,15 +49,22 @@ def __init__(self, message: str):
+            if executor == 'CMS AOD':
+                self._exe = cms_aod_executor()
+            else:
+                self._exe = atlas_xaod_executor()
```
Any reason not to catch the bad string here? So that if someone says "foo bar" we crash?

gordonwatts reviewed (changes requested):
Good as a draft!

gordonwatts committed:
Add protection

gordonwatts committed:
Fix up reference to requirement

gordonwatts committed:
Fix up how different images are built

- Normalize atlas vs cms naming
- Make docker file work properly

gordonwatts committed:
Sort out app.conf

`app.conf` is hard coded in several places, so best to just copy it from another source into that hardcoded path.

gordonwatts committed:
New version of func_adl_xAOD

Get the latest and greatest - with lots of fixes.

gordonwatts committed:
Bump to latest working version of xAOD

Now copies TTree to expected spot

gordonwatts committed:
Move to official version of code
